ITISFoundation/osparc-simcore
# 🔨 Enhances release monitor script

## What do these changes do?

_First round_ of enhancements on the `release-monitor` script.  The main goal is to move from a custom `.env` file to a `repo.config` file as inputs. The latter are curated env files in the `osparc-config` repo and each deploy has its own.

This is an initial PR that does not change the original functionality but rather defines a new `ReleaseSettings` to test the new concept before removing the legacy. 

### Highlights
1. Enhances CLI: uses commands instead of `action` options and improve checks on path
2. Moves code `release/monitor` to `release`
3. New command to display parsed and validated settings

### Suggestion 
I would like to use a shorter program name. Since this is about MOnitoring Osparc Deploy .. I was wondering something like `mood`. Then it would read as
```
mood settings ls
mood --env-file=.env sidecars ls
```
What do you think?

## Related issue/s

Maintenance

## How to test

Install
```
cd scripts/release
make devenv
monitor-release --help
```
Create a link or pass a path to any of the `repo.config`  (note that by doing that you are already selecting the `deploy` that your are targeting)
```cmd
ln -s /path/to/osparc-config-repo/deployments/osparc-master.speag.com/repo.config
```
List settings
```
monitor-release settings
```
will display loaded settings

## Dev Checklist


## DevOps Checklist
